Mobile version

razvanso opened the pull request:
Implements:
- Works section
- About section
- Skills lists, expandable via CSS only

Also adds local linting support (started by merging fix-local-linter-feedback branch into mobile_version)

razvanges committed:
Add Webhint

$ npm init
$ npm i -D hint@6.x
$ wget https://github.com/microverseinc/linters-config/blob/master/html-css/.hintrc

razvanges committed:
Add Stylelint

$ npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
$ wget https://github.com/microverseinc/linters-config/blob/master/html-css/.stylelintrc.json

razvanges committed:
Fix local linting configuration

Add script command for npm t | test | lint

Replaced .hintrc and .stylelintrc.json contents:
wget file doesn't work properly,
  because it needs to target raw.githubusercontent.com/...
  instead of the regular file path

Ignored Hint reports

razvanges committed:
Add Works section contents & images

razvanges committed:
Style Works section

razvanges committed:
Fix bottom padding of Works section

razvanges committed:
Format code

razvanges committed:
Refactor button alignment in Flex containers

razvanges committed:
Add About section with expandable skill lists

razvanges committed:
Fix most linting warnings

razvanges committed:
Refactor toggles to use SVGs instead of characters

MrBrN197 commented on index.html:
```diff
@@ -43,6 +43,142 @@ <h1>
         <li><a href="#"><img src="images/Vector.svg" alt="Vector"></a></li>
       </ul>
     </section>
+    <div class="works">
+      <article class="card-works">
+        <img class="card-image" src="images/works/project_01.main_snapshot.jpg" alt="Main snapshot of project 1">
+        <div class="card-details">
```
- [x] Please use a `section` gat for this `.works` section instead of using a generic `div` tag.

razvanso replied:
Thank you for your feedback, Brian!
The reason I changed from <section> to a regular <div> was a linting warning complaining about 'no heading for this section'. Can you please suggest how should I treat this issue? A hidden heading would be fine, or simply ignoring the linting warning is preferred?

MrBrN197 commented on index.html:
```diff
@@ -43,6 +43,142 @@ <h1>
+          <li><a href="#"><img src="images/Medium.svg" alt="Medium"></a></li>
+        </ul>
+        <a href="#" class="btn">Get my resume</a>
+        <section class="skill-list">
+          <input id="languages" class="toggle" type="checkbox" checked>
+          <label for="languages" class="toggle-label">Languages</label>
+          <ul class="collapsible-category">
+            <li><img src="images/works/JS.png" alt="JavaScript logo">JavaScript</li>
+            <li><img src="images/works/HTML.png" alt="HTML logo">HTML</li>
+            <li><img src="images/works/CSS.png" alt="CSS logo">CSS</li>
+          </ul>
+        </section>
+        <section class="skill-list">
+          <input id="frameworks" class="toggle" type="checkbox">
+          <label for="frameworks" class="toggle-label">Frameworks</label>
+          <ul class="collapsible-category">
+            <li><img src="images/works/HTML.png" alt="HTML logo">HTML</li>
+            <li><img src="images/works/CSS.png" alt="CSS logo">CSS</li>
+            <li><img src="images/works/JS.png" alt="JavaScript logo">JavaScript</li>
+          </ul>
+        </section>
+        <section class="skill-list">
+          <input id="skills" class="toggle" type="checkbox">
+          <label for="skills" class="toggle-label">Skills</label>
+          <ul class="collapsible-category">
+            <li><img src="images/works/CSS.png" alt="CSS logo">CSS</li>
+            <li><img src="images/works/JS.png" alt="JavaScript logo">JavaScript</li>
+            <li><img src="images/works/HTML.png" alt="HTML logo">HTML</li>
+          </ul>
+        </section>
```
- [x] Please wrap these three `.skill-list` sections using a `ul` and place each of them inside a `li` tag. the `.skill-list`s are a list of different skill sections so it might make more sense to have them inside a `ul`. I hope this makes sense.

razvanso replied:
Great feedback! Thanks

MrBrN197 reviewed (changes requested):
## Hi, @razvanso  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ Design matches Figma fantastically 👏.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

razvanges committed:
Incorporate feedback

Wraps skill lists into a container list
Use section instead of div as a Works container

razvanges committed:
Fix all known warnings

- Add hidden heading to Works section
- Refactor two sections into one

ShoiraTa reviewed (approved):
### **STATUS: APPROVED  ✔️** 
------
Hi @razvanso,  

Your project is complete🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

<img src="https://www.serialminds.com/wp-content/uploads/2015/11/loser.gif" width="400" height="250" />

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
